Create compatibility_table (#141)
commerce-docs/microsite-commerce-storefront: src/content/docs/get-started/release.mdx

---
title: Release information
description: Determine which versions of the drop-in components are compatible with each other and with Adobe Commerce.
---

> Adobe organizes storefront application releases into logical suite versions and follow a regular release schedule (approximately once a month). However, Adobe may release updates on demand (if necessary).

Reviewer: ```suggestion
Adobe organizes storefront application releases into logical suite versions and follow a regular release schedule (approximately once a month). However, Adobe may release updates as necessary.
```

> ## Compatibility Table

Reviewer: ```suggestion
## Compatibility table
```

---
title: Release information
description: Determine which versions of the drop-in components are compatible with each other and with Adobe Commerce.
---

> Adobe organizes storefront application releases into logical suite versions and follow a regular release schedule (approximately once a month). However, Adobe may release updates as necessary.

Reviewer 2: Is this the agreed on flow? Will we continue using `develop` past GA and merge into main once a month for these "suite releases"?

Reviewer 3: Is this about the Boilerplate? Each drop-in has its release strategy in place.

Author: Yes (doesn't necessarily have to be `develop`, but some branch, yeah). More details here: https://wiki.corp.adobe.com/display/COREENG/Post-GA+Drop-ins+Release+Plan.

## Compatibility table

The following table shows the compatibility between all of the storefront release items by suite version:

| Release item | Version | |
|---|---|---|

> | **Storefront suite** | 1.0.0 |

Reviewer 2: Where is the version noted in code? As tag in github on boilerplate repo? Just in the docs here? There's a package.json in boilerplate, so I guess we could bump that.

Infact if we want to do that, we need it as a step during release to commit the version bump to develop before merging

Author: Discussion in https://github.com/commerce-docs/microsite-commerce-storefront/pull/141#discussion_r1882475358.

| **Released** | December 2024 |
| **Adobe Commerce** | 2.4.7 |
| **Storefront compatibility package** | 4.7.0 |
| **Drop-in SDK** | 0.37.0 |

> | **Drop-in components** | Product Details: 1.0.0<br/>Cart: 1.0.0<br/>Checkout: 1.0.0<br/>User Auth: 1.0.0<br/>User Account: 1.0.0<br/>Orders: 1.0.0 |

Reviewer 2: In terms of "suite", and what we include here, I think you can just check the boilerplate package.json and list them here, including "@dropins/build-tools", @dropins/tools", for example.

https://github.com/hlxsites/aem-boilerplate-commerce/blob/develop/package.json#L28

Author: Fixed in https://github.com/commerce-docs/microsite-commerce-storefront/pull/141/commits/3b8df5cdcdd19a3ae0fa52faf2039669bdddb4b0

## Code changes

Every release includes changes to one or more codebases listed above. The scope of the change determines whether the MAJOR, MINOR, or PATCH number increases in the application's version according to [semantic versioning](https://semver.org/):

> Given a version number MAJOR.MINOR.PATCH, increment the:
> 
> MAJOR version when you make incompatible API changes  
> MINOR version when you add functionality in a backward compatible manner  
> PATCH version when you make backward compatible bug fixes  

> Note: The "Storefront suite" version is more of a marketing construct and does not conform to semantic versioning.

Reviewer 2: So...when we release a dropin and update `main` on boilerplate, should we expect to bump the suite version?

Author: No, not always. Package versions yes, and we'll update the package ranges

> ### Public API

Author: @fnhipster @sirugh @shiftedreality please review this section with an eye on:

1. Should we include functions coming from the SDK at this point? The SDK is not officially `1.0.0` yet but breaking the signatures of something like `initialize` or `render` will break customer sites. If we do include these functions, please enumerate the "public" ones we expect customers to use (some question marks in the following lines)
2. Anything that I missed?

Reviewer 2: I don't actually think we should include SDK in docs. It is not public. what is public is the _output_ packages of the SDK, including fetchGql, tools, event-bus, etc.

But I do think we should, like you say, include signatures of render/initialize/etc. those come from tools

In order to determine when we increment the MAJOR version we must define a Public API.

The Public API for drop-ins includes the following:

#### Integration APIs

1. Function signatures of public functions in Drop-in SDK tools
  * `fetch-graphql`
    * `setEndpoint`
    * `setFetchGraphQlHeader`
    * `removeFetchGraphQlHeader`
    * `setFetchGraphQlHeaders`
    * `fetchGraphQl`
    * `getConfig`

>   * `event-bus`?

Author: Are there functions from the `event-bus` that we should include?

Reviewer 2: The event bus has `on` and `emit`.

Reviewer 3: and `lastPaylod` https://git.corp.adobe.com/Commerce-Storefront/StorefrontSDK/blob/434bb967125b90ac02219a76a7def43fcc0d5b62/packages/event-bus/src/index.ts#L31

  * [Elsie exports](https://git.corp.adobe.com/Commerce-Storefront/StorefrontSDK/blob/82df5ab3b9f23314c2b0816628c928b43643f74b/packages/elsie/src/lib/index.ts)?

>     * Function signatures of `render` and `initialize`

Author: I listed at least `render` and `initialize` here ... are there others?

Reviewer 3: You can review the files exposed by `@dropins/tools` at https://www.npmjs.com/package/@dropins/tools?activeTab=code

`types/signatures are in /@dropins/tools/types/elsie/src/`

Author: Can you help me word this one @fnhipster ? I don't want to list everything there.

#### Slots

1. Default properties available to all slots:
  * Function signatures of functions `prependSibling`, `prependChild`, `replaceWith`, `appendChild`, `appendSibling`, `getSlotElement`, `onChange`
  * Structure of `dictionary` object
2. Function signatures of individual drop-in slots as defined elsewhere on this site
  * [Cart slots](https://experienceleague.adobe.com/developer/commerce/storefront/dropins/cart/slots/), for example

#### Drop-in design tokens

> 1. Drop-in design tokens defined by the [drop-in SDK](https://elsie.dev/?path=/docs/design-overview--overview)

Author: We probably don't want to actually link to this development documentation site. Not sure how best to enumerate all of these tokens otherwise.

Reviewer 4: We might be able to pull in a subset of the SDK docs via component that @bdenham developed in this repo: https://github.com/commerce-docs/dropins.dev

Reviewer 4: Actually, can we just link here @supernova-at?

https://experienceleague.adobe.com/developer/commerce/storefront/dropins/all/branding/#design-tokens

Author: Yes, perfect!
